install.sh

```diff
@@ -79,125 +77,133 @@ function backup_append() {
-
-  while ! [ $append_to_config ]; do
-    read -e -n 1 -r -p "Would you like to keep your $CONFIG_FILE and append bash-it templates at the end? [y/N] " choice
-    case $choice in
-    [yY])
-      backup_append
-      break
-      ;;
-    [nN]|"")
-      backup_new
-      break
-      ;;
-    *)
-      echo -e "\033[91mPlease choose y or n.\033[m"
-      ;;
-    esac
-  done
+	if [ -e "$HOME/$BACKUP_FILE" ]; then
+		echo -e "\033[0;33mBackup file already exists. Make sure to backup your .bashrc before running this installation.\033[0m" >&2
+		while ! [ $silent ]; do
+			read -e -n 1 -r -p "Would you like to overwrite the existing backup? This will delete your existing backup file ($HOME/$BACKUP_FILE) [y/N] " RESP
+			case $RESP in
+				[yY])
+					break
+					;;
+				[nN] | "")
+					echo -e "\033[91mInstallation aborted. Please come back soon!\033[m"
+					exit 1
+					;;
+				*)
+					echo -e "\033[91mPlease choose y or n.\033[m"
+					;;
+			esac
+		done
+	fi
+
+	while ! [ $append_to_config ]; do
+		read -e -n 1 -r -p "Would you like to keep your $CONFIG_FILE and append bash-it templates at the end? [y/N] " choice
+		case $choice in
+			[yY])
+				backup_append
+				break
+				;;
+			[nN] | "")
+				backup_new
+				break
+				;;
+			*)
+				echo -e "\033[91mPlease choose y or n.\033[m"
+				;;
+		esac
+	done
 elif [[ $append_to_config ]]; then
-  # backup/append
-  backup_append
+	# backup/append
+	backup_append
 elif [[ $silent ]] && ! [[ $no_modify_config ]]; then
-  # backup/new by default
-  backup_new
+	# backup/new by default
+	backup_new
 fi
 
 # Disable auto-reload in case its enabled
 export BASH_IT_AUTOMATIC_RELOAD_AFTER_CONFIG_CHANGE=''
 # Load dependencies for enabling components
-source "$BASH_IT/lib/composure.bash"
+# shellcheck source=./vendor/init.d/composure.bash
+source "$BASH_IT/vendor/init.d/composure.bash"
+# shellcheck disable=SC1090
+# shellcheck source=./lib/utilities.bash
 source "$BASH_IT/lib/utilities.bash"
 cite _about _param _example _group _author _version
+# shellcheck source=./lib/helpers.bash
 source "$BASH_IT/lib/helpers.bash"
 
-if [[ $interactive ]] && ! [[ $silent ]] ;
-then
-  for type in "aliases" "plugins" "completion"
-  do
-    echo -e "\033[0;32mEnabling $type\033[0m"
-    load_some $type
-  done
+if [[ $interactive ]] && ! [[ $silent ]]; then
+	for type in "aliases" "plugins" "completion"; do
+		echo -e "\033[0;32mEnabling $type\033[0m"
+		load_some $type
+	done
 else
-  echo ""
-  echo -e "\033[0;32mEnabling reasonable defaults\033[0m"
-  _enable-completion bash-it
-  _enable-completion system
-  _enable-plugin base
-  _enable-plugin alias-completion
-  _enable-alias general
+	echo ""
+	echo -e "\033[0;32mEnabling reasonable defaults\033[0m"
+	_enable-completion bash-it
+	_enable-completion system
+	_enable-plugin base
+	_enable-plugin alias-completion
+	_enable-alias general
 fi
 
 echo ""
 echo -e "\033[0;32mInstallation finished successfully! Enjoy bash-it!\033[0m"
+# shellcheck disable=SC2086
```

Comment:
why disable the warning here?

Reply:
Shellcheck was reading `'source "$HOME/$CONFIG_FILE"'` as an actual source and wanted "" around `$HOME`. It suggested `'source ""$HOME"/$CONFIG_FILE"'`.

Reply:
I could put the " around `$HOME` but it would not give the wanted string for the echo cmd

Reply:
I see, resolving